docs/specs/component.md

```diff
@@ -205,53 +200,72 @@ sending it, like the `prometheus_exporter` sink, SHOULD NOT publish this metric.
+      with relatively low cardinality because it will be used as a metric tag.
+      Examples would be syscall error code. Examples of values that should not
+      be used are raw error messages from `serde` as these are highly variable
+      depending on the input. Instead, these errors should be converted to an
+      error code like `invalid_json`.
+  * `error_type` - The type of error condition. MUST be one of the types listed
+    in the `error_type` enum list in the cue docs.
+  * `stage` - The stage at which the error occurred. MUST be one of `receiving`,
+    `processing`, or `sending`.
   * If any of the above properties are implicit to the specific error
     type, they MAY be omitted from being represented explicitly in the
     event fields. However, they MUST still be included in the emitted
     logs and metrics, as specified below, as if they were present.
 * Metrics
-  * MUST increment the `component_errors_total` counter by 1 with the defined
-    properties, except `message` as metric tags.
+  * If `recoverable` is `false`, MUST increment the `component_errors_total`
+    counter by 1. It should include the defined properties as tags, except for
+    `message` and `recoverable`.
   * MUST increment the `component_discarded_events_total` counter by the number
     of Vector events discarded if the error resulted in discarding (dropping)
-    acknowledged events. For sources, only increment this metric if incoming
-    events were consumed (and acknowledged if applicable) and discarded. The
-    metric MUST not include events that will be re-ingested. For sinks, this
-    means only incrementing this metric if the error resulted in the sink
-    dropping the events, and thus acknowledging them. Retried events MUST not be
-    included in the metric.
+    acknowledged events.
+    * For sources, only increment this metric if incoming events were consumed
+      (and acknowledged if applicable) and discarded. The metric MUST not
+      include events that will be re-ingested.
+    * For sinks, this means only incrementing this metric if the error resulted
+      in the sink dropping the events, and thus acknowledging them. Retried
+      events MUST not be included in the metric.
+* Logs
+  * If `recoverable` is `false`, MUST log a message at the `error` level.
+    Otherwise, if `recoverable` is `true`, MUST log a message at the `warn`
+    level.
+  * MUST include the defined properties as key-value pairs.
+  * SHOULD be rate limited to 10 seconds.
+
+#### ComponentEventsDropped
+
+*All components* that can drop events, **unrelated to an error**, MUST emit a
+`ComponentEventsDropped` event. Otherwise, if an event is dropped due to an
+error, then the error event itself MUST emit the appropriate logs and metrics
+as described in the [ComponentError](#ComponentError) section.
+
+* Metrics
+  * MUST increment the `component_discarded_events_total` counter by the number
+    of Vector events discarded.
```

Comment:
How will users differentiate between events dropped intentionally (like by the `filter` transform) and those dropped due to an error? Also, what tags should be applied? Should it have the same `stage` tag as the error `component_discarded_events_total` has?

Reply:
I think it's the presence of the `error_type` tag, right? I still think it's somewhat strange that the `component_discarded_events_total` has an `error_type` tag, but that's how it works now. So I think it's fine in the short term.

Reply:
🤔 that feels to me like it is overloading the `error_type` tag a bit. Maybe we could add a `reason` tag that is `error|intentional`?

Reply:
Hm, can you think of a case where we'd unintentionally drop data not due to an error?

The only one I can think of is an event that never matches any of the routes in the `route` transform, but that could be an error.

Reply:
Yeah, I was thinking transforms like `route`, `sample`, and `throttle`.

Reply:
Resolved and moved the discussion [here](https://github.com/vectordotdev/vector/pull/12822#discussion_r880782837).